### DISCUSSION TITLE
pixel analysis

### DISCUSSION THREAD

**Discussion by CA10C3SAR** [ImageMagick/ImageMagick#2283] [Help] [answered]
using this image as example:
![test](https://user-images.githubusercontent.com/6724006/87749420-05f01800-c7cf-11ea-8b40-ebd1902c368a.png)

the image is 256x256 suppose it starts at x0 y0 

this redline is just to illustrate:
![Photoshop_2020-07-17_01-54-10](https://user-images.githubusercontent.com/6724006/87750181-1c976e80-c7d1-11ea-8d5a-1d74957c8504.png)

x93 y35 is the first non-yellow pixel

is possible to use ImageMagick to return an array differentiating the yellow pixels than any other?

what I would need is something like:
0,0,0
[...]
89,35,0
90,35,0
91,35,0
92,35,0
93,35,1

0 to yellow pixels 1 to non-yellow

**Comment by fmw42** [comment]
Sorry, I do not understand the question. Also what is your exact IM version and platform, since syntax may vary.

**Reply by CA10C3SAR** [reply]
> Sorry, I do not understand the question. Also what is your exact IM version and platform, since syntax may vary.

ImageMagick-7.0.10-Q16 / Win10
What did you don't understand?

I'm trying to generate an array from this image, saving it as x,y,0 for yellow pixels and
x,y,1 for non-yellow

**Comment by snibgo** [comment]
You can make transparent all yellow (or non-yellow) pixels, then write these to "sparse-color:", which gives the text coordinates of non-transparent pixels. For example:
```
magick in.png -transparent yellow sparse-color:
```

**Comment by CA10C3SAR** [comment]
wow, you're the man @snibgo ! 
i got this as a result:

`magick in.png -transparent yellow sparse-color:`
![image](https://user-images.githubusercontent.com/6724006/87790625-3cea1c00-c817-11ea-8c07-233c71ae53e6.png)

how i could save the output to a txt file? if possible just the coordinates like 89,35, 90,36 etc

**Comment by snibgo** [comment, marked as answer]
You can write directly to a text file, or (slightly more slowly) redirect to a text file:
```
magick in.png -transparent yellow sparse-color:myfile.txt

magick in.png -transparent yellow sparse-color:>myfile.txt
```
Annoyingly, we can't change the unfriendly output from within IM. We can pipe to a tool that changes spaces to newlines. We could use Unix tools for other filtering. I prefer to use a Windows "for" loop, like this Windows BAT syntax.

```
(
  for /F "usebackq tokens=1,2 delims=," %%A in (`%IMG7%magick ^
    87749420-05f01800-c7cf-11ea-8b40-ebd1902c368a.png -transparent Yellow ^
    sparse-color: ^| tr " " "\n"`) do @echo %%A,%%B
)>x.txt

```
The first few lines of x.txt are:
```
93,35
92,36
93,36
74,37
83,37
```
